### DISCUSSION TITLE
Scripts and Logs being stored on Agents for multiple years

### DISCUSSION THREAD

**Discussion by fireball8931** [Ylianst/MeshCentral#5850] [Issues]
**Describe the bug**
Noticed that after using Mesh Central for a few years now, that the Mesh Agent folder will fill up with .ps1 and .txt files that appear to be scripts and logs of the scripts of commands I have sent to the agent over the years.
One machine which hadn't been reimaged in a couple of years had over 2 GB of these files.

**To Reproduce**
1. Install Mesh Central and Mesh agent across the network
2. Run Mesh Central normally for multiple years, including running scripts remotely.
3. Run DIR while in the Mesh Agent folder

**Expected behavior**
I did not expect the scripts and the logs to be locally stored.

**Server Software (please complete the following information):**
 - OS: Ubuntu
 - Virtualization: Runs directly on server
 - Network: WAN, SSL, multiple networks
 - Version: Agent: 1.1.16
 - Node: v12.18.2 (can't figure out how to get new version to kick in, will submit report for that)

**Remote Device (please complete the following information):**
 - Device: Desktop
 - OS:  Windows 10 22H2
 - Network: Remote over WAN
 - Current Core Version (if known): 3687806373

**Additional context**

**Your config.json file**
Not including, not sure how it's relevant to the problem.

This may be a security issue, as the logs may contain PII

**Comment by si458** [comment]
how are you running scripts?
are you using the plugin 'ScriptTask'?

**Comment by fireball8931** [comment]
I use the Action either from the device General Tab or the Group Action.
The only plugin I'm using is the WFH plugin (which I wish I could have the Azure checkbox checked by default)

Edit: If I need a script to run on a schedule, I have been using the Proactive Remediations in Intune.

**Comment by si458** [comment]
when we `run commands` we dont save anything to text files?
sample code for running the command line actions
```
mesh.cmdchild = require('child_process').execFile(process.env['windir'] + '\\system32\\cmd.exe', ['cmd'], options);
mesh.cmdchild.descriptorMetadata = 'UserCommandsShell';
mesh.cmdchild.stdout.on('data', function (c) { replydata += c.toString(); });
mesh.cmdchild.stderr.on('data', function (c) { replydata += c.toString(); });
mesh.cmdchild.stdin.write(data.cmds + '\r\nexit\r\n');
mesh.cmdchild.on('exit', function () { sendConsoleText(replydata); sendConsoleText("Run commands completed."); delete mesh.cmdchild; });
```
this simply runs cmd with your commands and exits.
the reply gets stored in memory and then sent back to the console of the device in the webui ?

edit:
do you have a sample of some output?
i can search phrases in the source code to see if maybe we done, but im 90% sure we dont save anything to text files?

**Comment by si458** [comment]
if you store scripts/files inside the 'My Files' tab, then they will be stored inside the `meshcentral-files` but thats only the file sourcecodes to run, not the outputs of the data from the machines?

**Comment by si458** [comment]
can you share your config.json? (hide secret info)
and where are the files  with the data?
on the remote device (mesh agent) or on the meshcentral server?

**Comment by si458** [comment]
if i look at the `ScriptTask` plugin sourcecode,they save txt and ps1 files on the remote agent devices so it might possibly be that
https://github.com/ryanblenis/MeshCentral-ScriptTask/blob/master/modules_meshcore/scripttask.js#L148-L205

EDIT: moving to a discussion as this isnt meshcentral issue

**Comment by fireball8931** [comment]
So this is interesting.
I checked the directory in the Mesh Agent folder on another machine and sorted by date. The files stop in July of 2023.
That must be around when I stopped using it.
Thanks for the insight.

It looks like that all the devices we have reimaged since July 2023 do not have these files.
